### ISSUE TITLE
Loss4 and Loss5 is increasing

### ISSUE THREAD

**Issue by modic23** [6lyc/CDNMF#1]
Hi,

I'm trying to run this model for identifying communities. Number of nodes are 3900 and number of features are 1024. I want to extract around 500 communities for understand the behavioural pattern. 
Facing a couple of issues:
1. While I'm able to run the code, I see the loss4 and loss5 is increasing with every epoch. Unable to understand why is this happening. 
2. Even though I'm changing the hyper params, my resulting output is not changing significantly. The communities are not breaking down into granular or small communities. 

Can you please help here?

Thanks

**Comment by 6lyc** [comment]
Hello,

Thank you for your interest in our model, and apologies for the delayed response—I was racing to meet the NeurIPS submission deadline. Please see my comments below:

1. **Regarding Q1**  
   In `Model/my_model.py` (line 206), the loss is defined as:  
   ```python
   loss = self.rec * (loss1 + loss2) 
        + self.conc * loss3 
        + self.r    * loss4 
        + self.negc * loss5
   ```

If you observe that `loss4` and `loss5` are not decreasing, try increasing their weighting coefficients (`self.r` and `self.negc`). In our experience, `loss5` is more important—so I recommend prioritizing adjustments to `self.negc` first.

2. **Regarding Q2**
   The total number of communities is controlled by the last element of both `pretrain_config['net_shape']` and `model_config['net_shape']`. If you want to decompose the network into smaller, more granular communities, increase these values—for example, set them to around **500**.

I hope this helps resolve your issues! If you have further questions or can share more details about your dataset & specific error logs, please feel free to continue the discussion.

Best regards,
*Yuecheng Li*
